src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filteredTodos, setFilteredTodos] = useState<string>('');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  useEffect(() => {
+    getTodos().then(setTodosList);
+  }, []);
+
+  let visibleTodos = [...todosList];
+
+  if (filteredTodos) {
+    switch (filteredTodos) {
+      case 'all':
+        visibleTodos = todosList;
+        break;
+      case 'active':
+        visibleTodos = visibleTodos.filter(todo => todo.completed === false);
+        break;
+      case 'completed':
+        visibleTodos = visibleTodos.filter(todo => todo.completed === true);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (searchQuery) {
+    const preparedQuery = searchQuery.toLowerCase();
+
+    visibleTodos = visibleTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  }
```

Comment:
1. You can rewrite this logic using only one filter method
2. I suggest you create separate function for this logic or put this code to useMemo